react-intervaler/src/components/Timer.jsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import Form from 'react-bootstrap/Form';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+
+const Timer = () => {
+  const [seconds, setSeconds] = useState(0);
+  const [minutes, setMinutes] = useState(0);
+  const [isActive, setIsActive] = useState(false);
+  const [secondsDisplayed, setSecondsDisplayed] = useState(0);
+  const [secondsInput, setSecondsInput] = useState(0);
+  const [minutesInput, setMinutesInput] = useState(0);
+
+  function toggle() {
+    setIsActive(!isActive);
+  }
+
+  function reset() {
+    setSeconds(0);
+    setSecondsDisplayed(0);
+    setMinutes(0);
+    setIsActive(false);
+  }
+
+  function setTimer() {
+    let timer = 0;
+    if (!minutesInput){ 
+      setMinutesInput(0);
+    }
+    timer = (minutesInput*60) + secondsInput;
+    setSeconds(timer)
+    setMinutes(Math.floor(timer/60));
+    setSecondsDisplayed(secondsDisplayed => (timer - (minutesInput*60)));
+    console.log(minutes)
+    console.log(seconds)
+    console.log(timer)
+  }
+
+  useEffect(() => {
```

Comment:
You can turn this entire useEffect into it's own custom hook. Then call it both in Timer and Stopwatch

Reply:
it differs between the two however, i guess i can rewrite it to take in a up or down or something from state??? maybe??

Reply:
It's good to turn hooks into something that is generic so that you can reuse them between components. It also lets you test then outside of the component that uses them.

End of day, we want to use hooks because it allows us to separate code that is presentational and code that contains "business logic" as well as test them independently of each other.